modules/patterns/pages/managing-monorepo-applications.adoc

```diff
@@ -0,0 +1,81 @@
+
+=== Changes that affect multiple components in the monorepo
+
+If the change spans across multiple components or the release process is expected to release multiple components at the same time, then the users can benefit from the xref:testing:integration/snapshots/group-snapshots.adoc[group Snapshot testing] functionality of the Integration service where all the updated component builds are combined into a single Snapshot and tested together.
+
+Having custom integration tests for this kind of testing can be enabled by using the `group` testing context in a custom IntegrationTestScenario. Refer to xref:testing:integration/choosing-contexts.adoc[the guide for choosing when to run integration test scenarios].
+
+== Push event testing considerations for monorepo applications
+
+While the group testing functionality allows us to determine how the final changes will look after they are merged, this functionality is unfortunately currently not directly available after the Pull Request is merged.
+
+As a result, after merging the Pull Request, the individual build pipelines will result in intermediate Snapshots which will not contain all the changes until the final build pipelineRun completes. These Snapshots can be considered as redundant in cases of monorepos, and releasing them would mean that only the partial change is put forward for release.
+
+To help with this problem, a custom IntegrationTestScenario pipeline can be created for `push` events (also see xref:testing:integration/choosing-contexts.adoc[the guide for choosing when to run integration test scenarios]) which checks if a Snapshot contains all the changes from the monorepo. If not, it fails the test and prevents the Snapshot from being auto-released, thus saving time and resources.
+
+=== Example Tekton task for checking if all Snapshot components match
+
+See the example `test-snapshot-component` Tekton task which checks if all Snapshot components belonging to the same repository were built from the same commit:
+
+[source,yaml]
+----
+tasks:
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: test-snapshot-component
+spec:
+  params:
+    - name: SNAPSHOT
+      description: The JSON string of the Snapshot under test
+  steps:
+    - name: test-snapshot-component
+      image: quay.io/konflux-ci/konflux-test:stable
+      workingDir: /workspace
+      env:
+        - name: SNAPSHOT
+          value: $(params.SNAPSHOT)
+        - name: SNAPSHOT_URL
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.annotations['pac.test.appstudio.openshift.io/source-repo-url']
+        - name: SNAPSHOT_SHA
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.annotations['pac.test.appstudio.openshift.io/sha']
+      script: |
+        #!/bin/sh
```

Comment:
`[[ ]]` is not supported in #!/bin/sh and we should use `#!/bin/bash` instead,  wdyt?

Reply:
```suggestion
        #!/bin/bash
```